pkg/controller/direct/logging/utils.go

```diff
@@ -84,6 +84,22 @@ func compareMetricDescriptors(kccObj *krm.LogmetricMetricDescriptor, apiObj *api
 	return reflect.DeepEqual(kccObj, convertAPItoKRM_MetricDescriptor(apiObj))
 }
 
+func validateImmutableFieldsUpdated(kccObj *krm.LogmetricMetricDescriptor, apiObj *api.MetricDescriptor) error {
+	actualMetricDescriptor := convertAPItoKRM_MetricDescriptor(apiObj)
+
+	modified := []string{}
+	if !reflect.DeepEqual(kccObj.MetricKind, actualMetricDescriptor.MetricKind) {
+		modified = append(modified, "metricDescriptor.MetricKind")
+	}
+	if !reflect.DeepEqual(kccObj.ValueType, actualMetricDescriptor.ValueType) {
```

Comment:
I think we should guard against kccObj being nil or actualMetricDescriptor being nil.  Maybe we do punt this "double-check" to 1.119, IDK

Reply:
I guess they can be nil? If both are nil then we consider it as "no changes on field".